osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/SetConfigurationObjectServiceDsmr4.java

```diff
@@ -0,0 +1,63 @@
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.configuration.service;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+import org.openmuc.jdlms.datatypes.BitString;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.utils.DlmsHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.Protocol;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagTypeDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationObjectDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GprsOperationModeTypeDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SetConfigurationObjectServiceDsmr4 extends SetConfigurationObjectService {
+
+    public SetConfigurationObjectServiceDsmr4(final DlmsHelper dlmsHelper) {
```

Comment:
will throw NPE at runtime when used.

Reply:
Nope.

Reply:
Can you explain? the constructor does not autowire the dlmshelper.

Reply:
Starting with Spring 4.3, if a class, which is configured as a Spring bean, has only one constructor, the Autowired annotation can be omitted and Spring will use that constructor and inject all necessary dependencies.

--> I did not know this feature ! so forget previous comments